### ISSUE TITLE
How do I parse only part of YAML into an object?

### ISSUE THREAD

**Issue by rcdailey** [aaubry/YamlDotNet#599]
Suppose I have the following YAML:

```yml
config_one:
  name: foo
  stuff: value

config_two:
  name: bar
  random: value
```

I want to selectively parse `config_one` into an object and I want `config_two` to be ignored:

```cs
class ConfigOne
{
  public string Name {get;set;}
  public string Stuff {get;set;}
}
```

How can I do this? The documentation is pretty lacking, or at least, it uses a lot of terminology that doesn't make much sense to me, and thus I was not able to search for this functionality.

**Comment by pensono** [comment]
This code should help you. The trick is to use the parser to scan to the point where you would like to deserialize.

https://dotnetfiddle.net/TMqvhX
```csharp
using System;
using System.IO;
using YamlDotNet.Core;
using YamlDotNet.Core.Events;
using YamlDotNet.Serialization;

var yaml = @"
config_one:
  name: foo
  stuff: value

config_two:
  name: bar
  random: value";


var parser = new Parser(new StringReader(yaml));
var deserializer = new DeserializerBuilder().Build();

parser.Consume<StreamStart>();
parser.Consume<DocumentStart>();
parser.Consume<MappingStart>();

while (parser.TryConsume<Scalar>(out var key))
{
	if (key.Value == "config_two")
	{
		var config = deserializer.Deserialize<dynamic>(parser);
		Console.WriteLine(config["random"]);
	} 
	else
	{
		parser.SkipThisAndNestedEvents();
		Console.WriteLine($"Skipped {key.Value}");
	}
}
```

**Comment by rcdailey** [comment]
Minor detail, but I found that because I'm *deserializing*, I have to use `UnderscoredNamingConvention.Instance` to get my code to recognize the YAML to C# property mappings. I don't know why this is contrary to many examples I'm seeing. My assumption was that direction (serialize/deserialize) affects the behavior. Deserializing means we have to tell the code to map underscored names `config_one` to whatever is there in the C# property, which is camel case: `ConfigOne`. I probably have this wrong but I'm basing this off of what works and what doesn't. If I use camel case enumerator as you have in your example, the YAML does not parse.

**Comment by rcdailey** [comment]
Thanks again for the solution. I changed my YAML slightly, but I still need to accomplish the same task (only parse `config_two`). How do you adapt your solution to work with `List<ConfigTwo>`?

```yml
config_one:
  name: foo
  stuff: value

config_two:
  - name: bar1
    random: value1
  - name: bar2
    random: value2
```

Doing this does not work:

```cs
var config = deserializer.Deserialize<List<ConfigTwo>>(parser);
```

**Comment by rcdailey** [comment]
Sorry for the false positive. This actually does work, but it breaks when `ConfigTwo` has a nullable enum member:

```cs
enum PossibleNames { Bar1, Bar2 }
class ConfigTwo
{
  public PossibleNames? Name { get; set; }
};
```

I made it nullable to express that this property in YAML is optional. You don't have to provide `name:` at all. I'm not sure if this is the appropriate representation of optional YAML properties mapped to enums.

**Comment by rcdailey** [comment]
Found the solution in #544. Hopefully that gets proper support by the library in the future. I'm closing now since my question is answered. Thanks again!

**Comment by pensono** [comment]
Should be fixed once #600 is merged :)
